AWSPinpoint/AWSPinpointNotificationManager.m

```diff
@@ -153,84 +153,87 @@ - (void)handleNotificationReceived:(UIApplication *) app
+        AWSDDLogError(@"Not valid Pinpoint push notification");
+        return;
+    }
+
+    NSDictionary *metadata = [self getMetadataFromUserInfo:userInfo];
+    [self addPinpointMetadataForEvent:pushNotificationEvent
+                     withMetadata:metadata];
     
     [self.context.analyticsClient recordEvent:pushNotificationEvent];
 }
 
 - (void)recordMessageOpenedEventForNotification:(NSDictionary *) userInfo
-                                 withIdentifier:(NSString *) identifier
-                           withApplicationState:(UIApplicationState) state {
+                                 withIdentifier:(NSString *) identifier {
     //User tapped on notification
-    AWSPinpointEvent *pushNotificationEvent = [self.context.analyticsClient createEventWithEventType:AWSEventTypeOpened];
-    
+    AWSPinpointEvent *pushNotificationEvent = [self buildEventFromUserInfo:userInfo
+                                                        withPushActionType:AWSPinpointPushActionTypeOpened];
+    if (!pushNotificationEvent) {
+        AWSDDLogError(@"Not valid Pinpoint push notification");
+        return;
+    }
     if (identifier) {
         [pushNotificationEvent addAttribute:identifier forKey:AWSAttributeActionIdentifierKey];
     }
     
-    [self addApplicationStateAttributeToEvent:pushNotificationEvent
-                         withApplicationState:state];
+
     [self.context.analyticsClient recordEvent:pushNotificationEvent];
 }
 
 #pragma mark - Helpers
-- (void)addCampaignMetadataForEvent:(AWSPinpointEvent *) event
-                   withNotification:(NSDictionary *) userInfo {
-    if (![AWSPinpointNotificationManager validCampaignPushForNotification:userInfo]) {
-        return;
-    }
-    NSDictionary *campaign = userInfo[AWSDataKey][AWSPinpointKey][AWSPinpointCampaignKey];
-    AWSDDLogVerbose(@"Adding campaign attributes to event[%@]: %@",event.eventType, campaign);
-    
-    for (NSString *key in [campaign allKeys]) {
-        [event addAttribute:campaign[key] forKey:key];
+- (void)addPinpointMetadataForEvent:(AWSPinpointEvent *) event
+                   withMetadata:(NSDictionary *) metadata {
+    for (NSString *key in [metadata allKeys]) {
+        [event addAttribute:metadata[key] forKey:key];
     }
 }
 
-- (void)addGlobalCampaignMetadataForNotification:(NSDictionary *) userInfo {
-    if (![AWSPinpointNotificationManager validCampaignPushForNotification:userInfo]) {
-        return;
-    }
-    
-    NSDictionary *campaign = userInfo[AWSDataKey][AWSPinpointKey][AWSPinpointCampaignKey];
-    AWSDDLogVerbose(@"Adding campaign global attributes: %@", campaign);
-    [self.context.analyticsClient setCampaignAttributes:campaign];
-    
-    for (NSString *key in [campaign allKeys]) {
-        [self.context.analyticsClient addGlobalAttribute:campaign[key] forKey:key];
+- (void)addGlobalPinpointMetadata:(NSDictionary *) metadata {
+    [self.context.analyticsClient setFeatureAttributes:metadata];
```

Comment:
metadata can now be null, but we were previously checking if it was valid?

Reply:
Will add the check